app/scripts/modules/core/src/presentation/hooks/useContainerClassNames.hook.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { usePrevious } from './usePrevious.hook';
+
+const { useEffect, useRef } = React;
+
+const getContainerElement = () => document.querySelector('.spinnaker-container');
+
+export const useContainerClassNames = (classNames: string[]) => {
```

Comment:
I agree with the sentiment 100%, and would prefer to avoid this becoming a commonplace thing for sure. I'm not totally clear on what a good permanent solution might look like though, so for now I'm just trying to wrap up the yucky DOM mutation inside something encapsulated and well tested so the complexity doesn't bleed out into other places.

The core requirement is that we need to _somehow_ allow a component anywhere in the tree to tell a container up at the top of the tree that it should have/not have a class name. There's a couple things that complicate it:
- I don't want to have any explicit coupling between a component like a `Modal` and some other container component beyond saying "I need to apply this class to whatever the 'container' is"
- Right now the component that renders that `spinnaker-container` element is in angular. This one is more easily solved/temporary

One thing I was pondering was whether we should have some kind of "presentation context" that can be a bag of state for the entire component tree all the way from the root, and allows sub-components to get information about the overall presentational state of the app + declaratively ask to change things like this — similar to the formik/validation context etc. I could imagine the concept of themes also falling into that spot too.